src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/GardenDataJson.kt

```diff
@@ -0,0 +1,152 @@
+    val visits: Map<String, Int>,
+    val completed: Map<String, Int>,
+    @SerializedName("total_completed") val totalCompleted: Int,
+    @SerializedName("unique_npcs_served") val uniqueNpcsServed: Int,
+)
+
+data class ComposterData(
+    val upgrades: APIComposterUpgrades,
+)
+
+data class APIComposterUpgrades(
+    val speed: Int,
+    @SerializedName("multi_drop") val multiDrop: Int,
+    @SerializedName("fuel_cap") val fuelCap: Int,
+    @SerializedName("organic_matter_cap") val organicMatterCap: Int,
+    @SerializedName("cost_reduction") val costReduction: Int,
+)
+
+data class ComposterUpgrade(
+    val upgrade: Int,
+    val items: Map<String, Int>,
+    val copper: Int,
+)
+
+data class GardenRepoJson(
+    @SerializedName("garden_exp") val gardenExperience: List<Int>,
+    @SerializedName("crop_milestones") val cropMilestones: Map<CropType, List<Int>>,
+    @SerializedName("visitors") val visitors: Map<String, VisitorRarity>,
+    val plots: Map<String, PlotData>,
+    @SerializedName("plot_costs") val plotCosts: Map<String, List<PlotCost>>,
+    @SerializedName("barn") val barn: Map<String, BarnSkin>,
+    @SerializedName("crop_upgrades") val cropUpgrades: List<Int>,
+    @SerializedName("composter_upgrades") val composterUpgrades: Map<String, Map<Int, ComposterUpgrade>>,
+    @SerializedName("composter_tooltips") val composterTooltips: Map<String, String>,
+)
+
+data class PlotData(
+    val name: String,
+    val x: Int,
+    val y: Int,
+)
+
+data class PlotCost(
+    val item: String,
+    val amount: Int,
+)
+
+data class BarnSkin(
+    val name: String,
+    val item: String,
+)
+
+data class EliteWeightJson(
+    val totalWeight: Double,
+    val cropWeight: Map<CropType, Double>,
+    val bonusWeight: Map<String, Double>,
+)
+
+enum class CropType(val itemId: String, val apiName: String, val displayName: String) {
+    WHEAT("ENCHANTED_HAY_BLOCK", "WHEAT", "Wheat"),
+    NETHER_WART("ENCHANTED_NETHER_STALK", "NETHER_STALK", "Nether Wart"),
+    SUGAR_CANE("ENCHANTED_SUGAR", "SUGAR_CANE", "Sugar Cane"),
+    CARROT("ENCHANTED_CARROT", "CARROT_ITEM", "Carrot"),
+    POTATO("ENCHANTED_POTATO", "POTATO_ITEM", "Potato"),
+    COCOA_BEANS("ENCHANTED_COCOA", "INK_SACK:3", "Cocoa Beans"),
+    PUMPKIN("ENCHANTED_PUMPKIN", "PUMPKIN", "Pumpkin"),
+    MELON("ENCHANTED_MELON", "MELON", "Melon"),
+    CACTUS("ENCHANTED_CACTUS_GREEN", "CACTUS", "Cactus"),
+    MUSHROOM("ENCHANTED_BROWN_MUSHROOM", "MUSHROOM_COLLECTION", "Mushroom"),
+    ;
+
+    companion object {
+        fun fromApiName(apiName: String): CropType? {
+            val fixedInput = apiName.uppercase().replace(" ", "_")
+            val fromApiName = values().firstOrNull { it.apiName == fixedInput }
+            if (fromApiName != null) return fromApiName
+            return values().firstOrNull { it.name == fixedInput }
+        }
+    }
+}
+
+enum class VisitorRarity(val displayName: String, var visits: Int = 0, var completed: Int = 0) {
```

Comment:
Storing this kind of state in here feels weird. This should probably be a map instead.

Reply:
Also just noticed that those statics are accessed in a thread. Double problematic